Upgrade allowed dependency ranges for PHP 8.5 support

Ocramius opened the pull request.

Ocramius committed:
Upgrade allowed dependency ranges for PHP 8.5 support

Ocramius commented:
We still need PCOV to be upstreamed, before merging this.

zll600 commented:
>  We still need PCOV to be upstreamed, before merging this.

Hi, @Ocramius , the php8.5-pcov should be released. I think we can continue the 8.5 support. 

The php8.5-pcov can be installed in my forked repo actions. FYI: https://github.com/zll600/BackwardCompatibilityCheck/actions/runs/19708464928/job/56462269978?pr=1

Ocramius committed:
Skip reporting indirect deprecations

Ocramius committed:
Attempt to suppress deprecation warnings in main CLI process

The STDERR of the main process should be stable, but PHP deprecations (and their flawed
design) make it really challenging to operate with them dependant on PHP defaults.

We therefore force them to be off.

Deprecations affecting us on PHP 8.5, right now:

* https://github.com/revoltphp/event-loop/issues/114
* https://github.com/revoltphp/event-loop/pull/110
* https://github.com/symfony/symfony/issues/61772

Ocramius commented:
Thanks @zll600 - now being released :-)

Ocramius commented:
Actually, need to revert and apply this to 8.16.x :facepalm:

zll600 commented:
@Ocramius Really Thanks for your work 🫡
